src/symmetric.rs

```diff
@@ -112,8 +112,9 @@ impl Drop for Plaintext {
 impl From<&[u8; 30]> for Plaintext {
     fn from(source: &[u8; 30]) -> Plaintext {
         let (M1, _) = encode_to_group(source);
-        let M2: RistrettoPoint = RistrettoPoint::hash_from_bytes::<Sha512>(source);
-        let m3: Scalar = Scalar::hash_from_bytes::<Sha512>(source);
+        let h = Sha512::default().chain(&source);
+        let M2: RistrettoPoint = RistrettoPoint::from_hash(h.clone().chain(b"M2"));
+        let m3: Scalar = Scalar::from_hash(h.chain(b"m3"));
 
         Plaintext { M1, M2, m3 }
```

Comment:
Breaking change. The old hash functions reused keying material while the new functions are independent.